modules/machineset-yaml-alibaba.adoc

```diff
@@ -0,0 +1,154 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-infrastructure-machinesets.adoc
+// * machine_management/creating_machinesets/creating-machineset-alibaba.adoc
+
+ifeval::["{context}" == "creating-infrastructure-machinesets"]
+:infra:
+endif::[]
+
+:_content-type: REFERENCE
+[id="machineset-yaml-alibaba_{context}"]
+= Sample YAML for a machine set custom resource on Alibaba Cloud
+
+This sample YAML defines a machine set that runs in a specified Alibaba Cloud zone in a region and creates nodes that are labeled with
+ifndef::infra[`node-role.kubernetes.io/<role>: ""`.]
+ifdef::infra[`node-role.kubernetes.io/infra: ""`.]
+
+In this sample, `<infrastructure_id>` is the infrastructure ID label that is based on the cluster ID that you set when you provisioned the cluster, and
+ifndef::infra[`<role>`]
+ifdef::infra[`<infra>`]
+is the node label to add.
+
+[source,yaml]
+----
+apiVersion: machine.openshift.io/v1beta1
+kind: MachineSet
+metadata:
+  labels:
+    machine.openshift.io/cluster-api-cluster: <infrastructure_id> <1>
+ifndef::infra[]
+    machine.openshift.io/cluster-api-machine-role: <role> <2>
+    machine.openshift.io/cluster-api-machine-type: <role> <2>
+  name: <infrastructure_id>-<role>-<region> <3>
```

Comment:
Similarly, here should be `<infrastructure_id>-<role>-<zone>`, you can refer to [Installer](https://github.com/openshift/installer/blob/4f774dabb36d2df7bca3c360a402feea19ae42af/pkg/asset/machines/alibabacloud/ machinesets.go#L48)
However, I found that there is a problem here, for example: for the regional ID `cn-hangzhou` and the availability zone ID `cn-hangzhou-k`, it will remove the prefix `<region ID>-`(cn-hangzhou-), generate name for `shtest0131-bfswt-worker-k`, but for region id `ap-southeast-6` and Availability zone id `ap-southeast-6a` this fails, apparently `ap-southeast-6a` has no prefix `ap-southeast-6-`, so I think the installer needs to fix using ZoneID, like: shtest0131-bfswt-worker-cn-hangzhou-k, shtest0131-bfswt-worker-ap-southeast-6a.

Do you have any suggestions? @kwoodson @jianli-wei

Reply:
The above are the problems that Installer needs to fix. For the documentation, I think a format can be agreed: `<infrastructure_id>-<role>-<zone>`